### PR TITLE
noarch recipe

### PR DESCRIPTION
**Pull request by mrakitin**
Here is a noarch recipe which successfully produced a package in https://github.com/nsls-ii-forge/curio-feedstock/pull/2 after discussion with @CJ-Wright in https://github.com/conda-forge/caproto-feedstock/issues/9#issuecomment-518434480.

Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mrakitin** [comment]
@conda-forge-admin, please rerender

**Comment by mrakitin** [comment]
Is it an expected behavior that the package is built with Python 2.7, but the tests ran with Python 3.7?

Also, here are a few errors, which didn't cause the build to fail:
```py
...
Installing collected packages: curio
  Compiling /tmp/pip-install-PNoJku/curio/curio/__init__.py ...
    File "/tmp/pip-install-PNoJku/curio/curio/__init__.py", line 19
      __all__ = [*errors.__all__,
                 ^
  SyntaxError: invalid syntax
  
  Compiling /tmp/pip-install-PNoJku/curio/curio/bridge.py ...
    File "/tmp/pip-install-PNoJku/curio/curio/bridge.py", line 41
      async def _asyncio_task(self):
              ^
  SyntaxError: invalid syntax
  
  Compiling /tmp/pip-install-PNoJku/curio/curio/channel.py ...
    File "/tmp/pip-install-PNoJku/curio/curio/channel.py", line 77
      async def __aenter__(self):
              ^
  SyntaxError: invalid syntax
  
  Compiling /tmp/pip-install-PNoJku/curio/curio/debug.py ...
    File "/tmp/pip-install-PNoJku/curio/curio/debug.py", line 18
      def __init__(self, *, level=logging.INFO, log=log, filter=None, **kwargs):
                          ^
  SyntaxError: invalid syntax
  
  Compiling /tmp/pip-install-PNoJku/curio/curio/file.py ...
    File "/tmp/pip-install-PNoJku/curio/curio/file.py", line 83
      async def read(self, *args, **kwargs):
              ^
  SyntaxError: invalid syntax
  
  Compiling /tmp/pip-install-PNoJku/curio/curio/io.py ...
    File "/tmp/pip-install-PNoJku/curio/curio/io.py", line 123
      def makefile(self, mode, buffering=0, *, encoding=None, errors=None, newline=None):
                                             ^
  SyntaxError: invalid syntax
  
  Compiling /tmp/pip-install-PNoJku/curio/curio/kernel.py ...
    File "/tmp/pip-install-PNoJku/curio/curio/kernel.py", line 85
      def __init__(self, *, selector=None, debug=None, activations=None):
                          ^
  SyntaxError: invalid syntax
  
  Compiling /tmp/pip-install-PNoJku/curio/curio/meta.py ...
    File "/tmp/pip-install-PNoJku/curio/curio/meta.py", line 115
      async def context():
              ^
  SyntaxError: invalid syntax
  
  Compiling /tmp/pip-install-PNoJku/curio/curio/monitor.py ...
    File "/tmp/pip-install-PNoJku/curio/curio/monitor.py", line 98
      async def monitor_task(self):
              ^
  SyntaxError: invalid syntax
  
  Compiling /tmp/pip-install-PNoJku/curio/curio/network.py ...
    File "/tmp/pip-install-PNoJku/curio/curio/network.py", line 23
      async def _wrap_ssl_client(sock, ssl, server_hostname, alpn_protocols):
              ^
  SyntaxError: invalid syntax
  
  Compiling /tmp/pip-install-PNoJku/curio/curio/promise.py ...
    File "/tmp/pip-install-PNoJku/curio/curio/promise.py", line 29
      async def set(self, data):
              ^
  SyntaxError: invalid syntax
  
  Compiling /tmp/pip-install-PNoJku/curio/curio/queue.py ...
    File "/tmp/pip-install-PNoJku/curio/curio/queue.py", line 53
      async def get(self):
              ^
  SyntaxError: invalid syntax
  
  Compiling /tmp/pip-install-PNoJku/curio/curio/side.py ...
    File "/tmp/pip-install-PNoJku/curio/curio/side.py", line 21
      async def _aside_term(task):
              ^
  SyntaxError: invalid syntax
  
  Compiling /tmp/pip-install-PNoJku/curio/curio/signal.py ...
    File "/tmp/pip-install-PNoJku/curio/curio/signal.py", line 138
      def __init__(self, *signos, maxsize=0, **kwargs):
                                        ^
  SyntaxError: invalid syntax
  
  Compiling /tmp/pip-install-PNoJku/curio/curio/socket.py ...
    File "/tmp/pip-install-PNoJku/curio/curio/socket.py", line 38
      async def create_connection(*args, **kwargs):
          ^
  SyntaxError: invalid syntax
  
  Compiling /tmp/pip-install-PNoJku/curio/curio/ssl.py ...
    File "/tmp/pip-install-PNoJku/curio/curio/ssl.py", line 31
      async def wrap_socket(sock, *args, do_handshake_on_connect=True, **kwargs):
          ^
  SyntaxError: invalid syntax
  
  Compiling /tmp/pip-install-PNoJku/curio/curio/subprocess.py ...
    File "/tmp/pip-install-PNoJku/curio/curio/subprocess.py", line 68
      async def wait(self):
              ^
  SyntaxError: invalid syntax
  
  Compiling /tmp/pip-install-PNoJku/curio/curio/sync.py ...
    File "/tmp/pip-install-PNoJku/curio/curio/sync.py", line 49
      async def wait(self):
              ^
  SyntaxError: invalid syntax
  
  Compiling /tmp/pip-install-PNoJku/curio/curio/task.py ...
    File "/tmp/pip-install-PNoJku/curio/curio/task.py", line 165
      async def _task_runner(self, coro):
              ^
  SyntaxError: invalid syntax
  
  Compiling /tmp/pip-install-PNoJku/curio/curio/thread.py ...
    File "/tmp/pip-install-PNoJku/curio/curio/thread.py", line 48
      async def _coro_runner(self):
              ^
  SyntaxError: invalid syntax
  
  Compiling /tmp/pip-install-PNoJku/curio/curio/traps.py ...
  SyntaxError: 'return' with argument inside generator (traps.py, line 82)
  
  Compiling /tmp/pip-install-PNoJku/curio/curio/workers.py ...
    File "/tmp/pip-install-PNoJku/curio/curio/workers.py", line 54
      async def run_in_executor(exc, callable, *args):
              ^
  SyntaxError: invalid syntax
  

Successfully installed curio-0.9
```
Is it expected?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -13,14 +13,14 @@ source:
...
-  number: 1001
-  script: python setup.py install --single-version-externally-managed --record record.txt
-  skip: True  # [win or py27]
+  noarch: python
+  number: 1002
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
 
 requirements:
   build:
     - python
```
Comment (CJ-Wright) [inline recipe/meta.yaml]:
```suggestion
    - python >=3.6
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -13,14 +13,14 @@ source:
...
-    - setuptools
+    - pip
   run:
     - python
```
Comment (CJ-Wright) [inline recipe/meta.yaml]:
```suggestion
    - python >=3.6
```

**Comment by CJ-Wright** [comment]
The tests can run with any valid configuration, thus it ran with 2.7. If you restrict the python version it should be fixed.

**Comment by mrakitin** [comment]
Thanks @CJ-Wright, I applied your suggestions. Let's see if we observe the issue mentioned in https://github.com/conda-forge/curio-feedstock/pull/7#issuecomment-518444580 now.

**Comment by mrakitin** [comment]
The build and test logs seem to be cleaner now: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=59918&view=logs&jobId=1bf226d3-0e2f-52d8-fa93-7d9e633347b3.

@danielballan [expressed concern](https://github.com/conda-forge/caproto-feedstock/issues/9#issuecomment-518614963) regarding making this package `noarch`, which makes sense even if the package does not have C/C++ extensions. What do @conda-forge/core folks think about it?

**Comment by CJ-Wright** [comment]
xref: https://github.com/conda-forge/staged-recipes/issues/4787

**Comment by scopatz** [comment]
@conda-forge-admin please rerender

**Comment by scopatz** [comment]
We tend to make things noarch if they can be.  It really reduces the maintence burden

**Comment by mrakitin** [comment]
Thanks for reviewing/merging @CJ-Wright and @scopatz!

**Pull request merged**